pkg/child/child.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package child provides the functionality to execute child command line processes.
+package child
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+)
+
+// Run executes a child process with the provided arguments.
+func Run(ctx context.Context, command string, args []string, workingDir string) ([]byte, []byte, int, error) {
+	path, err := exec.LookPath(command)
+	if err != nil {
+		return nil, nil, 1, fmt.Errorf("failed to locate command exec path: %w", err)
+	}
+
+	stdout := bytes.NewBuffer(nil)
+	stderr := bytes.NewBuffer(nil)
+
+	cmd := exec.CommandContext(ctx, path)
+	setSysProcAttr(cmd)
+	setCancel(cmd)
+
+	if workingDir != "" {
+		cmd.Dir = workingDir
+	}
+
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	cmd.Args = append(cmd.Args, args...)
+
+	// add small wait delay to kill subprocesses if context is canceled
+	// https://github.com/golang/go/issues/23019
+	// https://github.com/golang/go/issues/50436
+	cmd.WaitDelay = 2 * time.Second
+
+	select {
```

Comment:
Nit: comment. I'm assuming this is to check if the context has been cancelled before the command actually starts? If so, I'm not sure `cmd.ProcessState.ExitCode()` will be defined - you might need to check if `cmd.ProcessState != nil` first.

Reply:
Removed this as cmd.Start() handles this case...see test.